scalding-serialization/src/main/scala/com/twitter/scalding/serialization/OrderedSerialization.scala

```diff
@@ -32,9 +32,17 @@ trait OrderedSerialization[T] extends Ordering[T] with Serialization[T] {
    * the InputStreams is mutated to be the end of the record.
    */
   def compareBinary(a: InputStream, b: InputStream): OrderedSerialization.Result
+
+  /**
+   * This compares two InputStreams. After this call, the position in
+   * the InputStreams may or may not be at the end of the record.
+   */
+  def compareBinaryNoConsume(a: InputStream, b: InputStream): OrderedSerialization.Result = {
```

Comment:
I get really worried about how to compose methods like this that lack a strong contract. Also, I don't see that we ever call this.